Userland/Libraries/LibC/string.cpp

```diff
@@ -90,23 +90,23 @@ char* strndup(const char* str, size_t maxlen)
 // https://pubs.opengroup.org/onlinepubs/9699919799/functions/strcmp.html
 int strcmp(const char* s1, const char* s2)
 {
-    while (*s1 == *s2++)
-        if (*s1++ == 0)
-            return 0;
-    return *(const unsigned char*)s1 - *(const unsigned char*)--s2;
+    return strncmp(s1, s2, SIZE_MAX);
 }
 
 // https://pubs.opengroup.org/onlinepubs/9699919799/functions/strncmp.html
 int strncmp(const char* s1, const char* s2, size_t n)
 {
-    if (!n)
-        return 0;
-    do {
-        if (*s1 != *s2++)
-            return *(const unsigned char*)s1 - *(const unsigned char*)--s2;
-        if (*s1++ == 0)
+    const unsigned char* u1 = reinterpret_cast<const unsigned char*>(s1);
+    const unsigned char* u2 = reinterpret_cast<const unsigned char*>(s2);
+
+    for (size_t i = 0; i < n; i += 1) {
+        int cmp = static_cast<int>(u1[i]) - u2[i];
+        if (cmp != 0)
+            return cmp;
+        if (u1[i] == '\0' && u2[i] == '\0')
```

Comment:
The characters are the same at this point, so you only need to check one of them